mitosheet/src/mito/Mito.tsx

```diff
@@ -335,6 +335,23 @@ export const Mito = (props: MitoProps): JSX.Element => {
         loadPlotly()
     }, [])
 
+    /**
+     * This useEffect handles when the user creates a graph, then presses undo.
+     */
+    useEffect(() => {
+        if (uiState.currOpenTaskpane.type === TaskpaneType.GRAPH &&
+            analysisData.graphDataArray.length === 0) {
+            setUIState(prevUIState => {
+                return {
+                    ...prevUIState,
+                    currOpenTaskpane: {type: TaskpaneType.NONE},
+                    selectedTabType: 'data',
+                    selectedSheetIndex: 0
+                }
+            })
+        }
+    }, [analysisData.graphDataArray])
```

Comment:
Can we make this run when the array changes length? Feels like it might avoid potential race conditons